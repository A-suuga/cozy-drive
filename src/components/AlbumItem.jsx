import styles from '../styles/albumsList'

import React, { Component } from 'react'
// import classNames from 'classnames'
import { Link, withRouter } from 'react-router'
import { translate } from '../lib/I18n'

import { getPhotoLink } from '../actions/photos'

export class AlbumItem extends Component {
  constructor (props) {
    super(props)

    this.state = {
      isLoading: true
    }

    if (props.album && props.album.photosIds.length) {
      getPhotoLink(props.album.photosIds[0])
        .then(link => this.setState({
          url: link,
          isLoading: false
        }))
    } else {
      this.setState({
        url: '',
        isLoading: false
      })
    }
  }

  render () {
    const { t, album, router } = this.props
    const { isLoading, url } = this.state
    const parentPath = router.location.pathname
    return (
      !isLoading &&
        <div className={styles['pho-album']}>
          <Link
            to={`${parentPath}/${album._id}`}
            className={styles['pho-album-link']}
          >
            <img
              className={styles['pho-album-photo-item']}
              src={url}
            />
            <h2 className={styles['pho-album-title']}>{album.name}</h2>
            <h4 className={styles['pho-album-description']}>
              {t('Albums.album_item_description',
                {smart_count: album.photosIds.length})
              }
            </h4>
          </Link>
        </div>
    )
  }
}

export default translate()(withRouter(AlbumItem))
