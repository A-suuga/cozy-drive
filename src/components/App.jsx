import styles from '../styles/app'

import React from 'react'
import { translate } from '../lib/I18n'
import classNames from 'classnames'

import Sidebar from './Sidebar'

const App = ({ t, children }) => (
  <div className={classNames(styles['pho-wrapper'], styles['coz-sticky'])}>

    <Sidebar />

    <main className={styles['pho-content']}>
      { children }
    </main>
  </div>
)

export default translate()(App)
