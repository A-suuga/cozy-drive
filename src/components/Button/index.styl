@require 'cozy-ui'

@media (max-width: (1023/basefont)rem)
    .dri-btn--more
        display inline-flex

@media (max-width: (768/basefont)rem)
    .dri-btn--more
        border        0
        margin-right  0
        height        em(48px, 14px)
        opacity       .5