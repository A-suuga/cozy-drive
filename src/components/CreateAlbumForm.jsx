import styles from '../styles/createAlbumForm'

import React, { Component } from 'react'
import { translate } from '../lib/I18n'

import classNames from 'classNames'

class CreateAlbumForm extends Component {
  constructor () {
    super()
    this.reset()
  }

  reset () {
    this.setState({
      name: '',
      isSubmitDisabled: true,
      isBusy: false,
      hasError: false
    })
  }

  onSubmit (event, callback) {
    event.preventDefault()
    this.setState({ isBusy: true })
    return callback(this.state.name)
      .then(
        () => this.reset(),
        () => this.setState({
          hasError: true,
          isBusy: false
        })
      )
  }

  onNameChange (event) {
    const name = event.target.value.trim()
    this.setState({
      name: name,
      isSubmitDisabled: !name.length,
      hasError: false
    })
  }

  render ({ t, onSubmitNewAlbum }) {
    return (
      <form onSubmit={(event) => this.onSubmit(event, onSubmitNewAlbum)}>
        <label className={styles['coz-create-album-label']}>
          {t('Albums.create.inline_form.create_label')}
        </label>
        <div className={styles['coz-inline-form']}>
          <input
            className={classNames(styles['coz-input-text'], this.state.hasError && styles['error'])}
            type='text'
            name='album-name'
            onInput={(event) => this.onNameChange(event)}
            disabled={this.state.isBusy}
            placeholder={t('Albums.create.inline_form.placeholder')}
            value={this.state.name}
            />
          <button
            className={classNames('coz-btn', 'coz-btn--regular', styles['coz-btn'])}
            disabled={this.state.isSubmitDisabled || this.state.isBusy}
            aria-busy={this.state.isBusy}
            >
            {t('Albums.create.inline_form.create_button')}
          </button>
        </div>
      </form>
    )
  }
}

export default translate()(CreateAlbumForm)
