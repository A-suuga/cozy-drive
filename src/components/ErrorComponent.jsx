import styles from '../styles/errorComponent'

import React from 'react'
import { translate } from '../lib/I18n'

export const ErrorComponent = ({ t, errorType }) => {
  return (
    <div>
      {errorType === 'albums' &&
        <div className={styles['pho-error']}>
          <h2>{ t('Error.albums_title') }</h2>
          <button
            role='button'
            className='coz-btn coz-btn--regular'
            onClick={() => window.location.reload()}
          >
            {t('Error.refresh')}
          </button>
        </div>
      }
    </div>
  )
}

export default translate()(ErrorComponent)
