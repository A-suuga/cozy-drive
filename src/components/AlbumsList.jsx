import styles from '../styles/albumsList'

import React from 'react'

import Empty from '../components/Empty'
import AlbumItem from '../components/AlbumItem'

export const AlbumsList = ({ t, f, albums, onServerError }) => {
  if (albums.length === 0) {
    return <Empty emptyType='albums' />
  } else {
    return (
      <div className={styles['pho-album-list']}>
        {albums.map((a) => <AlbumItem album={a} key={a._id} onServerError={onServerError} />)}
      </div>
    )
  }
}

export default AlbumsList
