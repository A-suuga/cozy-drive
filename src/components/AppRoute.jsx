import React from 'react'
import { Route, Redirect } from 'react-router'

import Timeline from '../containers/Timeline'
import AlbumsView from '../containers/AlbumsView'
import AlbumPhotos from '../containers/AlbumPhotos'
import Viewer from '../containers/Viewer'

import App from './App'
export const ComingSoon = () => (<p style='margin-left: 2em'>Coming soon!</p>)

const AppRoute = (
  <Route component={App}>
    <Redirect from='/' to='photos' />
    <Route path='photos' component={Timeline}>
      <Route path=':photoId' component={Viewer} />
    </Route>
    <Route path='albums' component={AlbumsView}>
      <Route path=':albumId' component={AlbumPhotos}>
        <Route path=':photoId' component={Viewer} />
      </Route>
    </Route>
  </Route>
)

export default AppRoute
