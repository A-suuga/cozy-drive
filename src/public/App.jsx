/* global cozy */
import React, { Component } from 'react'
import { translate } from 'cozy-ui/react/I18n'

import PhotoBoard from '../components/PhotoBoard'
import Loading from '../components/Loading'
import ErrorComponent from '../components/ErrorComponent'
import {
  ALBUM_DOCTYPE
} from '../constants/config'

import styles from './index.styl'

class App extends Component {
  state = {
    name: null,
    photos: [],
    selected: [],
    loading: true,
    hasMore: false
  }

  onPhotoToggle = id => {
    this.setState(({ selected }) => {
      const idx = selected.findIndex(i => i === id)
      return {
        selected: idx === -1
          ? [...selected, id]
          : [...selected.slice(0, idx), ...selected.slice(idx + 1)]
      }
    })
  }

  onFetchMore = () => {

  }

  onDownload = () => {

  }

  async componentDidMount () {
    const { albumId } = this.props
    if (!albumId) {
      return this.setState({ error: 'Missing ID' })
    }
    const album = {
      _type: ALBUM_DOCTYPE,
      _id: albumId
    }
    try {
      const photosIds = await cozy.client.data.listReferencedFiles(album)
      const photos = await Promise.all(photosIds.map(cozy.client.files.statById))
      // const photosWithUrl = await Promise.all(photos.map(addUrl))
      const document = await cozy.client.data.find(ALBUM_DOCTYPE, albumId)
      this.setState(state => ({
        name: document.name,
        photos,
        loading: false
      }))
    } catch (ex) {
      return this.setState({ error: 'Sharing disabled', ex })
    }
  }

  render () {
    if (this.state.error) {
      return (
        <div className={styles['pho-public-layout']}>
          <ErrorComponent errorType={`public_album`} />
        </div>
      )
    }
    if (this.state.loading) {
      return (
        <div className={styles['pho-public-layout']}>
          <Loading loadingType='photos_fetching' />
        </div>
      )
    }
    const { t } = this.props
    const { name, photos, selected, hasMore } = this.state
    return (
      <div className={styles['pho-public-layout']}>
        <div className={styles['pho-content-header']}>
          <h2 className={styles['pho-content-title']}>{name}</h2>
          <div className={styles['pho-toolbar']} role='toolbar'>
            <div className='coz-desktop'>
              <button
                role='button'
                className={['coz-btn', 'coz-btn--secondary', styles['pho-public-download']].join(' ')}
                onClick={this.onDownload}
              >
                {t('Toolbar.album_download')}
              </button>
            </div>
          </div>
        </div>
        <PhotoBoard
          lists={[{ photos }]}
          selected={selected}
          showSelection={selected.length !== 0}
          onPhotoToggle={this.onPhotoToggle}
          hasMore={hasMore}
          onFetchMore={this.onFetchMore}
        />
        {this.renderViewer(this.props.children)}
      </div>
    )
  }

  renderViewer (children) {
    if (!children) return null
    return React.Children.map(children, child => React.cloneElement(child, {
      photos: this.state.photos
    }))
  }
}

export default translate()(App)
