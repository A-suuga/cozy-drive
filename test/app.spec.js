'use strict'

/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import { mockT } from './lib/I18n'
import { I18n } from '../src/lib/I18n'
import { App } from '../src/components/App'

describe('App component only', () => {
  it('should be mounted correctly', () => {
    const component = renderer.create(
      <App t={mockT} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('I18n with App component', () => {
  beforeEach(() => {
    jest.resetModules()
  })

  it('should be mounted correctly in en lang', () => {
    const component = renderer.create(
      <I18n lang='en'>
        <App />
      </I18n>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should be mounted correctly in fr lang', () => {
    const component = renderer.create(
      <I18n lang='fr'>
        <App />
      </I18n>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
