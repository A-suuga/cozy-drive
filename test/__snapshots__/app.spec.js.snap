exports[`App component only should be mounted correctly 1`] = `
<div
  className="pho-wrapper coz-sticky">
  <aside
    className="coz-sidebar">
    <nav>
      <ul
        className="coz-nav">
        <li
          className="coz-nav-item">
          <a
            className="pho-cat-photos"
            onClick={[Function]}
            style={Object {}}>
            Photos
          </a>
        </li>
        <li
          className="coz-nav-item">
          <a
            className="pho-cat-albums"
            onClick={[Function]}
            style={Object {}}>
            Albums
          </a>
        </li>
        <li
          className="coz-nav-item">
          <a
            className="pho-cat-shared"
            onClick={[Function]}
            style={Object {}}>
            Shared by me
          </a>
        </li>
        <li
          className="coz-nav-item">
          <a
            className="pho-cat-trash"
            onClick={[Function]}
            style={Object {}}>
            Trash
          </a>
        </li>
      </ul>
    </nav>
  </aside>
  <main
    className="pho-content" />
  <div
    className="pho-alerter" />
</div>
`;

exports[`I18n with App component should be mounted correctly in en lang 1`] = `null`;

exports[`I18n with App component should be mounted correctly in fr lang 1`] = `null`;
